analyzers/src/SonarAnalyzer.CSharp/Rules/SillyMathematicalComparison.cs

```diff
@@ -0,0 +1,107 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class SillyMathematicalComparison : SonarDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S2198";
+        private const string MathComparisonMessage = "Comparison to this constant is useless; the constant is outside the range of type '{0}'";
+        private const string ConstantComparisonMessage = "Don't compare constant values";
+
+        private static readonly DiagnosticDescriptor MathComparisonRule = DescriptorFactory.Create(DiagnosticId, MathComparisonMessage);
+        private static readonly DiagnosticDescriptor ConstantComparisonRule = DescriptorFactory.Create(DiagnosticId, ConstantComparisonMessage);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MathComparisonRule, ConstantComparisonRule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterNodeAction(
+                CheckBinary,
+                SyntaxKind.GreaterThanExpression,
+                SyntaxKind.GreaterThanOrEqualExpression,
+                SyntaxKind.LessThanExpression,
+                SyntaxKind.LessThanOrEqualExpression,
+                SyntaxKind.EqualsExpression,
+                SyntaxKind.NotEqualsExpression);
+
+        private static void CheckBinary(SonarSyntaxNodeReportingContext context)
+        {
+            var binary = (BinaryExpressionSyntax)context.Node;
+
+            var constantLeft = binary.Left.FindConstantValue(context.SemanticModel);
+            var constantRight = binary.Right.FindConstantValue(context.SemanticModel);
+
+            if (constantLeft is not null)
+            {
+                if (constantRight is not null) // both are constants
+                {
+                    context.ReportIssue(Diagnostic.Create(ConstantComparisonRule, binary.GetLocation()));
+                }
+                else // left is constant
+                {
+                    CheckOneSide(context, constantLeft, binary.Right);
+                }
+            }
+            else if (constantRight is not null) // right is constant
+            {
+                CheckOneSide(context, constantRight, binary.Left);
+            }
+        }
+
+        private static void CheckOneSide(SonarSyntaxNodeReportingContext context, object constant, SyntaxNode other)
+        {
+            if (ConversionHelper.TryConvertWith(constant, Convert.ToDouble, out var doubleConstant)
+                && context.SemanticModel.GetTypeInfo(other).Type is { } typeSymbolOfOther
+                && TryGetRange(typeSymbolOfOther, out var min, out var max)
+                && (doubleConstant < min || doubleConstant > max))
+            {
+                var typeName = context.SemanticModel.GetTypeInfo(other).Type.ToDisplayString();
```

Comment:
Use ToMinimalDisplayString. It takes `using` into account and fully qualifies the name only if needed. It also respects `static using` and using aliases.
```suggestion
                var typeName = typeSymbolOfOther.ToMinimalDisplayString(...);
```